File: go/trunk_maps.go
package main

import (
	"fmt"
	"os"
)

func main() {
	ex := os.Getenv("EXOPHONE")
	if ex == "" {
		ex = os.Getenv("DID_NUMBER")
	}
	if ex == "" {
		fmt.Println("Error: set EXOPHONE or DID_NUMBER")
		os.Exit(1)
	}
	q := map[string]string{"exophone": ex}
	if v := os.Getenv("TRUNK_SID"); v != "" {
		q["trunk_sid"] = v
	}
	fmt.Println("Trunk map lookup...")
	get("/trunk-maps", q)
}

File: python/trunk_maps.py
#!/usr/bin/env python3
"""Trunk map lookup: which trunk is tied to a virtual number (exophone). Requires DID_NUMBER or EXOPHONE."""
from _client import get
import os

ex = os.getenv("EXOPHONE") or os.getenv("DID_NUMBER")
if not ex:
    print("Error: set EXOPHONE or DID_NUMBER")
    exit(1)

q = {"exophone": ex}
if os.getenv("TRUNK_SID"):
    q["trunk_sid"] = os.getenv("TRUNK_SID")

print("Trunk map lookup...")
get("/trunk-maps", query=q)
